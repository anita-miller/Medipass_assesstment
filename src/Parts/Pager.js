import React from 'react'
import Button from '@material-ui/core/Button'

const Pager = (props) => {
    return(
        <div>
            {props.pageOffset === 1 ? null : (
            <Button 
              size="medium" 
              onClick ={props.prevPage}
              color= "primary"
              variant="contained"
            >
              Previous
            </Button>
            )}
            {props.currentPageLength < 5 ? null : (
            <Button 
              size="medium" 
              onClick ={props.nextPage}
              color= "secondary"
              variant="contained"
            >
        Next
            </Button>
            )}

        </div>
    )
}

export default Pager