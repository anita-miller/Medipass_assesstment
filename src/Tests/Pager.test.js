import React from 'react';
import Pager from '../Parts/Pager';
import renderer from 'react-test-renderer';


test('Pager component renders a snapshot', () =>{
    const pageLength = 5;
    const offset = 2;
    const component = renderer.create(
        <Pager
          currentPageLength={pageLength}
          pageOffset={offset}  
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();



});