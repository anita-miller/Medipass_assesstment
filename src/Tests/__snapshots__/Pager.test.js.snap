// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager component renders a snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Previous
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </button>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Next
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </button>
</div>
`;
